java/code/src/com/suse/manager/webui/utils/salt/custom/FilesDiffResult.java

```diff
@@ -47,7 +48,7 @@ public Map getPChanges() {
      * @return Object dataType's class
      */
     public <R> R getPChanges(Class<R> dataType) {
-        return JsonParser.GSON.fromJson(this.pchanges, dataType);
+        return JsonParser.GSON.fromJson(pchanges.orElse(getChanges()), dataType);
```

Comment:
@lucidd what do you think about this? Would it maybe make more sense to do this the other way round, that is use `StateApplyResult` if possible and fall back to `FilesDiffResult` only if `pchanges` is there? How could that be implemented?

Reply:
According to the example in the description no since old salt had `changes` set to an empty map so it would always be there just useless in case of old salt.